sync-catalog tok8s lowercase service names

lkysow opened the pull request:
When syncing from Consul to Kubernetes, lowercase the service names so
they're valid Kubernetes services.
Fixes https://github.com/hashicorp/consul-k8s/issues/110

lkysow commented on catalog/from-consul/source.go:
```diff
@@ -67,10 +69,40 @@ func (s *Source) Run(ctx context.Context) {
 
-			if !k8s {
-				services[s.Prefix+name] = fmt.Sprintf("%s.service.%s", name, s.Domain)
+			// Consul service names have no constraints, they can have any
```
I'm not sure if this should be done here or when we're about to create the service in Kubernetes in sink.go.

I put it here because in sink.go, all the maps use the service name as keys so I was worried they'd get out of sync

ishustava replied:
I think sink.go is a more appropriate place to do it since it seems to be more "k8s-aware".

lkysow commented:
@ishustava after looking at this again I've greatly simplified the implementation:
1. `source.go` doesn't know anything about the lowercasing, I moved that logic into sink (as suggested)
1. in sink.go, all we do is intercept the services in `SetServices()` and lowercase them. This way we're only ever dealing with lowercase names (and keys) inside sink.go
1. I removed the logging that we're lowercasing the services because it would be logged every sync period and also because it will be obvious that they have been lowercased simply by looking at Kubernetes
1. I removed the detection and logging that the name is invalid (in the case of special characters that are allowed in Consul but not in Kube for example) because during the Create() call to Kubernetes, we'll just get the error there and the registration will fail.
